book/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+
+if(MSVC)
```

Comment:
Is there a reason to protect it like this? Non-IDEs only build if ALL is set and other IDEs like Xcode May benefit too.

Reply:
generally not needed for most linux builds so in our other applications we just protect things like this, to not add overhead on CMAKE that isn't necessary.  I took it off.

Reply:
As for using a glob, I generally avoid it.  it can get a little annoying when you are adding things and cmake won't regenerate the project, whereas with a listing of all files that isn't a problem.   I made a it a glob but left a list of the readme and summary so if more are added they can be added in there.

Reply:
Glob should not be used for the main build system, but if CMake is a secondary build system (such as for a Python package that has setup.py as the primary system, or for an extra feature like this one), then it's quite useful - as it is easy to forget to keep changes updated everywhere. CONFIGURE_DEPENDS really helps make it more useful, but since docs is a "fake" target, there is no reason to expect a "--build" should also check this and reconfigure.